### ISSUE TITLE
How to configure monolog handler

### ISSUE THREAD

**Issue by mesilov** [getsentry/sentry-php#829]
I have a problem with configuring sentry with monolog usage

in documentation example they use old  raven client 
https://docs.sentry.io/clients/php/integrations/#capturing-errors 
```php
$client = new Raven_Client('https://<key>@sentry.io/<project>');

$handler = new Monolog\Handler\RavenHandler($client);
$handler->setFormatter(new Monolog\Formatter\LineFormatter("%message% %context% %extra%\n"));

$monolog->pushHandler($handler);
```
In other part of documentation we use new client
i try to configure monolog with new client, but can't see any events in sentry

```php
$arConfig = ['dsn' => 'https://...f1ad@sentry.io/1477408'];
$options = new Options($arConfig);
$transport = new HttpTransport($options, HttpAsyncClientDiscovery::find(), MessageFactoryDiscovery::find());
$builder = ClientBuilder::create($arConfig);
$builder->setTransport($transport);
Hub::getCurrent()->bindClient($builder->getClient());
$handlers[] = new \Sentry\Monolog\Handler(Hub::getCurrent(), Logger::ERROR);
$logger->pushHandler(new Monolog\Handler\WhatFailureGroupHandler($handlers));

```

**Comment by ste93cry** [comment]
When you set a custom transport you have to take care of adding also all the plugins that make the authentication works in the HTTP client

https://github.com/getsentry/sentry-php/blob/e2d99f1cf0158f5758b46e5e461052703a752d09/src/ClientBuilder.php#L292-L295

**Comment by GiorgioStokje** [comment]
This took me longer to figure out than I care to admit. It's been a while since your question, but perhaps it can help others searching.

Going with the basics, the new Monolog handler is used as follows:
```php
$logger = new \Monolog\Logger('logger_name');
$client = \Sentry\ClientBuilder::create(['dsn' => SENTRY_DSN])->getClient();
$handler = new \Sentry\Monolog\Handler(new \Sentry\State\Hub($client));
$logger->pushHandler($handler);
```

Then send a message as follows:
```php
$logger->error('Message', $context);
```

Context is passed along as an optional array:
```php
['exception' => $e, 'extra' => ['some' => 'content'], 'tags' => ['sentry' => 'tags']]
```

**Comment by ste93cry** [comment]
Closing this since you solved the issue yourself. Feel free to reopen if you need more help

**Comment by andreasschroth** [comment]
Please update your documentation properly. It's horrible. Why isn't there any example of how to instantiate the Monolog handler correctly?

**Comment by ste93cry** [comment]
getsentry/sentry-docs#1104 is tracking this issue and is waiting on a feature that will be available in `2.3` that will make it easier to instantiate and use the handler.

>Sorry for my harsh words, but this is not just open source, but a company and we pay for it

I cannot speak for the company behind Sentry, but I agree that the documentation is for sure not optimal/well organized. However, as you said this is open source so as I contribute (and probably it's my fault to not have documented the features I developed) you can do the same by updating the documentation if you think that something needs adjustments :wink:

**Comment by fnagel** [comment]
> Context is passed along as an optional array:
> 
> ['exception' => $e, 'extra' => ['some' => 'content'], 'tags' => ['sentry' => 'tags']]

Please note: this is no longer possible, see https://github.com/getsentry/sentry-php/pull/1068/files
